### DISCUSSION TITLE
Preset not found

### DISCUSSION THREAD

**Discussion by marvinfischenich** [renovatebot/renovate#16581] [Request Help]
Hi guys, my setup on self-hosted GitLab looks like this:

```
renovate-runner
|_ renovate-runner

renovate
|_ renovate-config
|_|_ default.json

apps
|_ base-image-update
|_|_ base-image update
|_|_|_ renovate.json
```

`renovate-runner/renovate-runner` is the Bot's location.

`default.json` holds the presets for every repo:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "platform": "gitlab",
  "includeForks": true,
  "dependencyDashboard": true,
  "hostRules": [
    {
      "matchHost": "...",
      "username": "...",
      "encrypted": {
        "password": "..."
      }
    }
  ],
  "packageRules": [
    {
      "matchDatasources":"pypi",
      "matchPackagePatterns": ["^..."],
      "registryUrls":[
        "..."
      ]
    },
    {
      "matchDatasources":"pypi",
      "matchLanguages":"python",
      "excludePackagePatterns": ["^..."],
      "registryUrls":[
        "..."
      ]
    }
  ]
}

```

the `renovate.json` looks like this
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "onboardingConfig": {
    "extends": [
      "renovate/renovate-config"
      ]
  }
}

```

Unfortunately something seems to be configured incorrect.
Renovate returns
` INFO: Repository is a fork and not manually configured - skipping (repository=apps/base-image-update/base-image-update)`

When i paste the content from `renovate/renovate-config/default.json` into `apps/base-image-updage/base-image-update/renovate.json` the bot is working properly.

How do i have to reference the `default.json` correctly?

Thanks for you support!

**Comment by PhilipAbed** [comment]
we usually put the `default.json` in a newly made repository.
then you can reference it with 
` "extends": ["github>{repoOwner}/{repoName}:default"] `
:default if the file name is `default.json`

see for example:
this config repo : https://github.com/whitesource/merge-confidence
we use `"extends": ["github>whitesource/merge-confidence:beta"]`

**Reply by PhilipAbed** [reply]
@marvinfischenich  for docs about referencing check this out: https://docs.renovatebot.com/config-presets/#gitlab

**Reply by marvinfischenich** [reply]
Hi @PhilipAbed, thanks for your reply but sorry i don't get it.
If i put a `default.json` in every newly made repo, what do i need presets then for?
In this case i already have all settings i want in my new repo.
The `default.json` holds all presets, right?

From my understanding i have a newly created repo abc.
This repo abc has a file `renovate.json` which references to a `default.json` for example in `renovate/renovate-config`.
When the developer is fine with all presets in `default.json` he just doesn't touch the `renovate.json` in his repo abc.
If he wants to change something he can modify his `renovate.json`.

What am i getting wrong?

**Reply by marvinfischenich** [reply]
> @marvinfischenich for docs about referencing check this out: https://docs.renovatebot.com/config-presets/#gitlab

since my gitlab is selfhosted i have to stick to [https://docs.renovatebot.com/config-presets/#self-hosted-git](url) instead, right?

**Reply by PhilipAbed** [reply]
i have never seen the config file in a different folder than the default folders before so.. i would follow the docs

**Comment by rarkins** [comment]
You should have one repo which contains the central preset in `default.json`. Then in all other repos you use `renovate.json` and then reference that central repo.

**Reply by marvinfischenich** [reply]
The central preset sits in
```
renovate
|_ renovate-config
|_|_ default.json
```

all other repos have a `renovate.json` for example:
```
apps
|_ base-image-update
|_|_ base-image update
|_|_|_ renovate.json
```

my `renovate.json` looks like this:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "onboardingConfig": {
    "extends": [
      "renovate/renovate-config"
      ]
  }
}
```

But that's not working.
Do you have an idea why?

**Reply by rarkins** [reply]
onboardingConfig is not a valid option for renovate.json

**Reply by marvinfischenich** [reply]
Before i tried the renovate.json with
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["renovate/renovate-config"]
}
```

but then i also get
`Repository is a fork and not manually configured - skipping (repository=apps/base-image-update/base-image-update)`

Then i found `If you are self hosting you must add "onboardingConfig": { "extends": ["config:base"] } to your bot's config.` [https://docs.renovatebot.com/config-presets/#how-to-use-preset-configs](url)

**Reply by rarkins** [reply]
"your bot's config" means the config.js file. Instead you added it to repo config
